scipy/spatial/transform/__init__.py

```diff
@@ -0,0 +1,43 @@
+"""
+========================================================
+Spatial Transformations (:mod:`scipy.spatial.transform`)
+========================================================
+
+.. currentmodule:: scipy.spatial.transform
+
+This package implements various spatial transformations. For now,
+only rotations are supported.
+
+Rotations in 3 dimensions
+-------------------------
+This package supports `Rotation` transforms with the following representations
+
+* Quaternions
+* Discrete Cosine Matrices
+* Euler angles
+* Axis angle representation
+
+It also implements some useful algorithms such as
+
+* Quaternion slerp
+* Quaternion spline
+* Rotation estimation (Wahba's problem)
+
+Class Reference
+---------------
+
+.. autosummary::
+    :toctree: generated/
+
+    Rotation
+
+"""
+from __future__ import division, print_function, absolute_import
+
+from .rotation import *
+
+__all__ = [s for s in dir() if not s.startswith('_')]
```

Comment:
Here you might as well just do `__all__ = ['Rotation']`. The `*` import above will import many things without leading `_` that we do not want in `__all__`, such as `np`, `re`, etc. And even if you change that import, then `division`, `print_function`, `absolute_import` will still be in `__all__` (and the code here will have the problem that if someone else does such a `*` import in the future they will have this problem).

Although the explicit list has some redundancy, it does not suffer from these problems.